File: src/main/java/aginsun/kingdoms/server/TaleOfKingdoms.java
package aginsun.kingdoms.server;

import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.SidedProxy;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
import net.minecraftforge.fml.common.event.FMLServerStartingEvent;

@Mod(modid = "taleofkingdoms", name = "Tale of Kingdoms", version = "1.0.12")
public final class TaleOfKingdoms
{
    @Mod.Instance("taleofkingdoms")
    public static TaleOfKingdoms instance;

    @SidedProxy(clientSide = "aginsun.kingdoms.client.ClientProxy", serverSide = "aginsun.kingdoms.server.ServerProxy")
    public static ServerProxy proxy;

    @Mod.EventHandler
    public void pre(FMLPreInitializationEvent e)
    {
        proxy.pre(e);
    }

    @Mod.EventHandler
    public void init(FMLInitializationEvent e)
    {
        proxy.init(e);
    }

    @Mod.EventHandler
    public void starting(FMLServerStartingEvent e)
    {
        proxy.starting(e);
    }
}

File: src/main/java/kingdoms/server/TaleOfKingdoms.kt
package kingdoms.server

import cpw.mods.fml.common.Mod
import cpw.mods.fml.common.Mod.EventHandler
import cpw.mods.fml.common.Mod.Instance
import cpw.mods.fml.common.SidedProxy
import cpw.mods.fml.common.event.FMLInitializationEvent
import cpw.mods.fml.common.event.FMLPreInitializationEvent
import cpw.mods.fml.common.event.FMLServerStartingEvent

@Mod(modid = "tok", name = "Tale of Kingdoms", version = "1.0.14")
class TaleOfKingdoms
{
    companion object
    {
        @Instance("tok")
        lateinit var instance: TaleOfKingdoms

        @SidedProxy(clientSide = "kingdoms.client.ClientProxy", serverSide = "kingdoms.server.ServerProxy")
        lateinit var proxy: ServerProxy
    }

    @EventHandler
    fun pre(e: FMLPreInitializationEvent)
    {
        proxy.pre(e)
    }

    @EventHandler
    fun init(e: FMLInitializationEvent)
    {
        proxy.init(e)
    }

    @EventHandler
    fun starting(e: FMLServerStartingEvent)
    {
        proxy.starting(e)
    }
}